### ISSUE TITLE
development-defaults.ini is missing some data that breaks things

### ISSUE THREAD

**Issue by khyperia** [magfest/ubersystem#2353]
When testing for #2350, I had to add a few items in `development-defaults.ini` that were causing various 500 errors or inability to do things. A big one was being able to get a volunteer to the shift signup page. Below is a description of the changes I did, and I'm making an issue instead of a PR since some of them are questionable or could be done better - I can throw a PR if desired, too. Ping @kitsuta.

---

`enums.{job_location,job_interest}` had no values in them, so when a volunteer was created, they couldn't be assigned to a department, and the check for "is assigned to a department" when logging in failed, and things didn't work. Also got 500 errors when accessing `http://host/uber/jobs/` due to `IndexError: c.JOB_LOCATION_OPTS[0][0]` being missing, things were generally broken, yadda yadda.

```ini
[enums]

[[job_location]]
test_department = "Test Department"

[[job_interest]]
interested_in_testing = "Interested In Testing"
```

Following that, once a department was created, some check somewhere was exploding due to `dept_head_checklist.{creating_shifts,assigned_volunteers}` being missing:

```ini
[dept_head_checklist]

[[creating_shifts]]
deadline = "2030-01-01"
description = "Create shifts"
path = "/jobs/index?location={department}"

[[assigned_volunteers]]
deadline = "2030-01-01"
description = "Assigned shifts"
path = "/jobs/staffers?location={department}"
```

Finally, `integer_enums.shirt` needed a valid shirt size for the volunteer checklist to be completed.

```ini
[integer_enums]
[[shirt]]
'shirty shirt' = 1
```

---

Also, for registering, the prereg deadline default was in the past (may want to change configspec instead? maybe change other default dates too in configspec?)

```ini
[dates]
prereg_takedown = "2030-01-01"
```

A questionable addition (to put into dev-defualts) I did was:

```ini
[dates]
badge_price_waived = "2000-01-01"

[badge_prices]
initial_attendee = 0
```

because I didn't want to deal with working around paying for registration (I think there's a Stripe test cc of 4242-... but it's a lot easier to just have it be free) - also I'm not sure which one actually made it free, but at this point I didn't care and just wanted it to work :stuck_out_tongue:

---

There's almost certainly other defaults that should be added, but those are the ones I specifically ran into. We might want to inspect `configspec.ini` and make sure any key in there and not in dev-defaults is actually OK to not be in dev-defaults.

**Comment by kitsuta** [comment]
I looked into this, thinking it'd be a pretty simple fix. It's not. :(

While we can probably fix the dates without too much trouble, all the enums are missing for good reason -- you can't get rid of them once you add them. So if we added a job location of (test_department, "Test Department"), events could change the display name but could never get rid of that variable.

@RobRuana has proposed adding the variables you found missing to https://github.com/magfest/production-config/blob/master/development.yaml. This would allow deploys done via vagrant to bypass these issues, but would make events reliant on MAGFest's config repo. That's not ideal, but it might be what we have to do for the time being.

**Comment by RobRuana** [comment]
Before changing https://github.com/magfest/production-config/blob/master/development.yaml, I'd want to make sure that isn't being used for our production deploys. But yeah...

I'm also open to any other proposals that will fix this for non-MAGFest organizations that are using rams.

**Comment by kitsuta** [comment]
Yeah, I think it would involve us doing massive refactors for comparatively little benefit. Documenting adding config as part of setting up certain features would be better and much faster!.

Eventually we want to stop relying on these config files for 'app config' anyway, so in the very very long run it's not worth it to do too much work.
